# Continuando con comandos en Terminal desde Linux

2. Comandos Intermedios

wc - Esto nos sirve para contar las lineas, letras, palabras y bytes que tiene el texto o archivo

In [3]:
!wc optd_aircraft_backup.csv

  458  1068 18612 optd_aircraft_backup.csv


esto significa 458 lineas, 1068 palabras y 18612 letras

head - Esto imprime las primeras 10 lineas del texto o archivo, cuando utilizas "head -n K" imprime las primeras K lineas, cuando utilizas "head -c K" imprime los primeros K bytes

In [5]:
!head optd_aircraft_backup.csv

iata_code,manufacturer,model,iata_group,iata_category,icao_code,nb_engines,aircraft_type
100,Fokker,100,100,2J,F100,2,J
141,BAe,146-100,146,4J,B461,4,J
142,BAe,BAE Systems 146-200 Passenger,146,4J,B462,4,J
143,BAe,146-300,146,4J,B463,4,J
146,BAe,146,,,,,
14F,BAe,146 Freighter (-100/200/300QT & QC),,,,,
14X,BAe,146 Freighter (-100QT & QC),14F,4J,B461,4,J
14Y,BAe,146 Freighter (-200QT & QC),14F,4J,B462,4,J
14Z,BAe,146 Freighter (-300QT & QC),14F,4J,B463,4,J


In [6]:
!head -n 20 optd_aircraft_backup.csv

iata_code,manufacturer,model,iata_group,iata_category,icao_code,nb_engines,aircraft_type
100,Fokker,100,100,2J,F100,2,J
141,BAe,146-100,146,4J,B461,4,J
142,BAe,BAE Systems 146-200 Passenger,146,4J,B462,4,J
143,BAe,146-300,146,4J,B463,4,J
146,BAe,146,,,,,
14F,BAe,146 Freighter (-100/200/300QT & QC),,,,,
14X,BAe,146 Freighter (-100QT & QC),14F,4J,B461,4,J
14Y,BAe,146 Freighter (-200QT & QC),14F,4J,B462,4,J
14Z,BAe,146 Freighter (-300QT & QC),14F,4J,B463,4,J
19D,Airbus,A319,,,,,
310,Airbus,A310,,,,,
312,Airbus,A310-200,310,2J,A310,2,J
313,Airbus,A310-300,310,2J,A310,2,J
318,Airbus,A318,32S,2J,A318,2,J
319,Airbus,A319,32S,2J,A319,2,J
31F,Airbus,A310F,,,,,
31W,Airbus,A319 (Sharklets),,,,,
31X,Airbus,A310-200F,31F,2J,A310,2,J
31Y,Airbus,A310-300F,31F,2J,A310,2,J


In [8]:
!head -c 1000 optd_aircraft_backup.csv

iata_code,manufacturer,model,iata_group,iata_category,icao_code,nb_engines,aircraft_type
100,Fokker,100,100,2J,F100,2,J
141,BAe,146-100,146,4J,B461,4,J
142,BAe,BAE Systems 146-200 Passenger,146,4J,B462,4,J
143,BAe,146-300,146,4J,B463,4,J
146,BAe,146,,,,,
14F,BAe,146 Freighter (-100/200/300QT & QC),,,,,
14X,BAe,146 Freighter (-100QT & QC),14F,4J,B461,4,J
14Y,BAe,146 Freighter (-200QT & QC),14F,4J,B462,4,J
14Z,BAe,146 Freighter (-300QT & QC),14F,4J,B463,4,J
19D,Airbus,A319,,,,,
310,Airbus,A310,,,,,
312,Airbus,A310-200,310,2J,A310,2,J
313,Airbus,A310-300,310,2J,A310,2,J
318,Airbus,A318,32S,2J,A318,2,J
319,Airbus,A319,32S,2J,A319,2,J
31F,Airbus,A310F,,,,,
31W,Airbus,A319 (Sharklets),,,,,
31X,Airbus,A310-200F,31F,2J,A310,2,J
31Y,Airbus,A310-300F,31F,2J,A310,2,J
320,Airbus,A320-100/200,32S,2J,A320,2,J
321,Airbus,A321-100/200,32S,2J,A321,2,J
32A,Airbus,A320 (sharklets),32S,2J,A320,2,J
32B,Airbus,A321 (sharklets),32S,2J,A321,2,J
32C,Airbus,A318 (sharklets),,,,,
32D,Airbus,A319 (sharklets),,,,,

tail - Esto imprime las ultimas 10 lineas del texto o archivo, cuando utilizas "head -n K" imprime las ultimas K lineas, cuando utilizas "head -c K" imprime los ultimos K byte

In [9]:
!tail optd_aircraft_backup.csv

TU3,Tupolev,Tu-134,TU3,2J,T134,2,J
TU5,Tupolev,Tu-154,TU5,3J,T154,3,J
VCV,Vickers,Viscount,,,,,
VCX,Vickers,VC-10,,,,,
WWP,IAI,1124 Westwind,WWP,2J,WW24,2,J
YK2,Yakovlev / Jakovlev,Yak 42,YK2,3J,YK42,3,J
YK4,Yakovlev / Jakovlev,Yak 40,YK4,3J,YK40,3,J
YN2,HAMC / Harbin,Y12,YN2,2T,Y12,2,T
YN7,Xian,Yunshuji Y7,YN7,2T,AN24,2,T
YS1,NAMC,YS-11,YS1,2T,YS11,2,T


In [10]:
!tail -n 20 optd_aircraft_backup.csv

SU7,Sukhoi,Superjet 100-75,SU1,2J,ZZZZ,2,J
SU9,Sukhoi,Superjet 100-95,SU1,2J,SU95,2,J
SWF,Fairchild-Swearingen,SA226 Freighter,SWF,2T,*,2,T
SWM,Fairchild-Swearingen,SA26 / SA226 / SA227 Metro / Merlin,SWM,2T,*,2,T
SY8,Shaanxi,Y-8,SY8,4T,AN12,4,T
T20,Tupolev,Tu-204 / Tu-214,T20,2J,T204,2,J
T2F,Tupolev,Tu-204 Freighter,T2F,2J,T204,2,J
T34,Tupolev,Tu-334,T34,2J,T334,2,J
TBM,DAHER-SOCATA,TBM-700,TBM,1T,TBM7,1,T
TRN,Surface Equipment,Train,TRN,S,0000,,S
TU3,Tupolev,Tu-134,TU3,2J,T134,2,J
TU5,Tupolev,Tu-154,TU5,3J,T154,3,J
VCV,Vickers,Viscount,,,,,
VCX,Vickers,VC-10,,,,,
WWP,IAI,1124 Westwind,WWP,2J,WW24,2,J
YK2,Yakovlev / Jakovlev,Yak 42,YK2,3J,YK42,3,J
YK4,Yakovlev / Jakovlev,Yak 40,YK4,3J,YK40,3,J
YN2,HAMC / Harbin,Y12,YN2,2T,Y12,2,T
YN7,Xian,Yunshuji Y7,YN7,2T,AN24,2,T
YS1,NAMC,YS-11,YS1,2T,YS11,2,T


In [11]:
!tail -c 1000 optd_aircraft_backup.csv

D.330,SH3,2T,SH33,2,T
SH6,Shorts,SD.360,SH6,2T,SH36,2,T
SHB,Shorts,SC-5 Belfast,,,,,
SHS,Shorts,SC-7 Skyvan,SHS,2T,SC7,2,T
SSC,Aerospatiale / SNIAS,Concorde,,,,,
SU1,Sukhoi,Superjet 100,,,,,
SU7,Sukhoi,Superjet 100-75,SU1,2J,ZZZZ,2,J
SU9,Sukhoi,Superjet 100-95,SU1,2J,SU95,2,J
SWF,Fairchild-Swearingen,SA226 Freighter,SWF,2T,*,2,T
SWM,Fairchild-Swearingen,SA26 / SA226 / SA227 Metro / Merlin,SWM,2T,*,2,T
SY8,Shaanxi,Y-8,SY8,4T,AN12,4,T
T20,Tupolev,Tu-204 / Tu-214,T20,2J,T204,2,J
T2F,Tupolev,Tu-204 Freighter,T2F,2J,T204,2,J
T34,Tupolev,Tu-334,T34,2J,T334,2,J
TBM,DAHER-SOCATA,TBM-700,TBM,1T,TBM7,1,T
TRN,Surface Equipment,Train,TRN,S,0000,,S
TU3,Tupolev,Tu-134,TU3,2J,T134,2,J
TU5,Tupolev,Tu-154,TU5,3J,T154,3,J
VCV,Vickers,Viscount,,,,,
VCX,Vickers,VC-10,,,,,
WWP,IAI,1124 Westwind,WWP,2J,WW24,2,J
YK2,Yakovlev / Jakovlev,Yak 42,YK2,3J,YK42,3,J
YK4,Yakovlev / Jakovlev,Yak 40,YK4,3J,YK40,3,J
YN2,HAMC / Harbin,Y12,YN2,2T,Y12,2,T
YN7,Xian,Yunshuji Y7,YN7,2T,AN24,2,T
YS1,NAMC,YS-11,YS1,2T,YS11,2,T


find - Esto busca archivos en directorio, utilize "find -type f" para buscar f(archivo),d(directorio). utiliza "find -name nombre" para encontrar por nombre, "find -perm 777" para encontrar archivos con todos los permisos, "find -not nombre" para buscar archivos que no tengan nombre, "find -size KM" para buscar archivos que pesan hasta K megas, "find (ruta)  -exec (ls -la {} \;)" para buscar en la ejecucion ls -la de la ruta (ruta)

In [15]:
!find -type f -name "optd*"

./optd_aircraft_backup.csv
./optd_aircraft.csv


In [58]:
!find ~ -type f -size +100M

/home/dsc/Data/challenge/searches.csv.bz2
/home/dsc/Data/challenge/bookings.csv.bz2
/home/dsc/anaconda3/pkgs/mkl-2018.0.1-h19d6760_4.tar.bz2


In [57]:
!find -perm a+r

./optd_aircraft.csv
./.git/objects/5b/39ea74ad5443c29840ee649cafc3bce0b2cc91
./.git/objects/9f/90bc286a355e394cebd55ad82e81455f5e1bfa
./.git/objects/9f/669cc0ebb32ceaa9ed051d234b77b3386df269
./.git/objects/88/1cd62446ae3ff4fa27a893251f1b8d40bd89ab
./.git/objects/a1/552c8ff26f391c3c6eb02e09c1f372b41f6205
./.git/objects/3b/18e512dba79e4c8300dd08aeb37f8e728b8dad
./.git/objects/ac/4e4b440fc6369e99a27004b565cafdced38bea
./.git/objects/32/a3e157db2aff37cbcc61885f142feadc0d95b3
./.git/objects/b1/ccf13e7cde3012ad7f746d77a6bb25da34de04
./.git/objects/c8/b1517391b7f023d39bb734b10c1c9664dfcd00
./.git/objects/c2/5462e9d16bceaf8c278244b9929541f3768835
./.git/objects/8c/ac0686d5e8a922bef8e5a20ccc63415cffb2ed
./.git/objects/c3/d96464cdd712f1ffdac40082e290ce85cb2fb0
./.git/objects/67/ecf55f952be4195c8f7dfd9902e14fdf78ff80
./.git/objects/6a/b5488185c7770f9d20fd6601501ad9efbdb757
./.git/objects/10/b231e699ac2dbde4796f4dc4225d383b1bba56
./.git/objects/78/cc4be754192cb256c41ccc57b7b7c401003c50
./.git/obje

In [34]:
!find -empty

./.git/branches
./.git/refs/tags
./.git/objects/pack
./.git/objects/info
./copy_directory


In [49]:
!find ~ -type f -size +100M -exec ls -la {} \;

-rwxr-x--- 1 dsc dsc 483188920 mar 13  2018 /home/dsc/Data/challenge/searches.csv.bz2
-rwxr-x--- 1 dsc dsc 554970628 mar 13  2018 /home/dsc/Data/challenge/bookings.csv.bz2
-rw-r--r-- 1 dsc dsc 193625096 feb  6  2018 /home/dsc/anaconda3/pkgs/mkl-2018.0.1-h19d6760_4.tar.bz2
